0000-product.md

```diff
@@ -0,0 +1,470 @@
+    repeated PropertyValues properties = 4; 
+} 
+```
+
+Validation requirements:
+
+* If a product with product_id does not exist the transaction is invalid.  The
+* signer of the transaction must be an agent in the Pike state and must
+belong to an organization in Pike state, otherwise the transaction is invalid.
+* The owner in the product must match the organization that the agent belongs
+* to, otherwise the transaction is invalid.  
+* The agent must have the permission can_update_prouduct for the organization,
+otherwise the transaction is invalid.
+* The new properties must be valid for the product_namespace. 
+* For example, if the product is GS1 product, its properties must only contain properties that are included in the GS1 Schema. If it includes a property not in the GS1 Scheme the transaction is invalid.
+
+The properties in the product will be swapped for the new properties and the
+updated product will be set in state.
+
+The inputs for ProductUpdateAction must include:
+
+* Address of the Agent submitting the transaction Address of the Organization
+* the Product is being updated for Address of the Product Namespace Schema the
+* product’s properties must match Address of the Product to be updated
+
+The outputs for ProductUpdateAction must include:
+
+* Address of the Product updated
+
+### ProductDeleteAction
+
+ProductDeleteAction removes an existing product from state. The transaction
+should be submitted by an agent, identified by its signing key, acting on behalf
+of the organization that corresponds to the org_id in the product being updated.
+(Organizations and agents are defined by the Pike smart contract.)
+
+``` 
+message ProductDeleteAction { 
+    enum Product_Namespace { 
+        UNSET_NAMESPACE = 0;
+        GS1 = 1; 
+    }
+    // product_namespace and product_id are used in deriving the state address
+    Product_Namespace product_namespace = 1; 
+    string product_id = 2; 
+} 
+```
+
+If the grid setting grid.product.allow_delete is set to false, this transaction
+is invalid. The default value for grid.product.allow_delete is true. This
+setting is stored using the Sawtooth Settings smart contract, more information
+can be found
+[here](https://sawtooth.hyperledger.org/docs/core/releases/latest/transaction_family_specifications/settings_transaction_family.html).
+
+Validation requirements:
+
+* If a product with product_id does not exist the transaction is invalid.  
+* The signer of the transaction must be an agent in the Pike state and must
+belong to an organization in Pike state, otherwise the transaction is invalid.
+* The owner in the product must match the organization that the agent belongs to, otherwise the transaction is invalid.  
+* The agent must have the permission “can_delete_product” for the organization,
+otherwise the transaction is invalid.
+
+The inputs for ProductDeleteAction must include:
+
+* Address of the Agent submitting the transaction Address of the Organization
```

Comment:
This bullet section is messed up. Several points are now running together and do not make sense.